File: pangram/pangram.cpp
#include <iostream>
#include <unordered_set>
#include <algorithm>

using namespace std;

/**
 * Проверяет, является ли строка панграммой
 * @param sentence Входная строка
 * @return true если строка содержит все буквы алфавита, иначе false
 */

bool isPangram(const string& sentence) {
    unordered_set<char> letters;
    for (char c : sentence) {
        if (isalpha(c)) {
            letters.insert(tolower(c));
        }
    }
    return letters.size() == 26;
}

int main() {
    cout << boolalpha << isPangram("thequickbrownfoxjumpsoverthelazydog") << endl; // true
    cout << boolalpha << isPangram("whatdoesthefoxsay") << endl; // false
    return 0;
}

File: pangram/pangram.go
package main

import (
	"fmt"
	"strings"
)

// IsPangram проверяет, является ли строка панграммой
func IsPangram(sentence string) bool {
	lowerSentence := strings.ToLower(sentence)
	for c := 'a'; c <= 'z'; c++ {
		if !strings.ContainsRune(lowerSentence, c) {
			return false
		}
	}
	return true
}

func main() {
	fmt.Println(IsPangram("thequickbrownfoxjumpsoverthelazydog")) // true
	fmt.Println(IsPangram("whatdoesthefoxsay")) // false
}